homeassistant/components/coolmaster/binary_sensor.py

```diff
@@ -20,8 +21,13 @@ async def async_setup_entry(
     async_add_entities: AddEntitiesCallback,
 ) -> None:
     """Set up the CoolMasterNet binary_sensor platform."""
-    async_add_entities_for_platform(
-        hass, config_entry, async_add_entities, CoolmasterCleanFilter
+    info = hass.data[DOMAIN][config_entry.entry_id][DATA_INFO]
+    coordinator = hass.data[DOMAIN][config_entry.entry_id][DATA_COORDINATOR]
+    async_add_entities(
+        [
+            CoolmasterCleanFilter(coordinator, unit_id, info)
+            for unit_id in coordinator.data
+        ]
```

Comment:
I *think* this works without the conversion to a list here and below.

```suggestion
    async_add_entities(
        CoolmasterCleanFilter(coordinator, unit_id, info)
        for unit_id in coordinator.data
```

Reply:
Done.